lib/protobuf/decoder.ex

```diff
@@ -198,6 +198,10 @@ defmodule Protobuf.Decoder do
     end
   end
 
+  defp decode_varints(n, type, acc) when is_integer(n) do
+    [Wire.to_proto(type, n) | acc]
+  end
+
```

Comment:
This seems to be the fix. We already get an integer here, not a binary.

Reply:
It seems strange that we get an integer when all other clauses expect an integer.